Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp

```diff
@@ -197,6 +211,223 @@ void FiniteDifferenceSolver::EvolveBCartesian (
+        Array4<Real> const &Rhoy = Rhofield[1]->array(mfi);
+        Array4<Real> const &Rhoz = Rhofield[2]->array(mfi);
+        amrex::Array4<amrex::Real> const &lx = edge_lengths[0]->array(mfi);
+        amrex::Array4<amrex::Real> const &ly = edge_lengths[1]->array(mfi);
+        amrex::Array4<amrex::Real> const &lz = edge_lengths[2]->array(mfi);
+        amrex::Array4<amrex::Real> const &Sx = face_areas[0]->array(mfi);
+        amrex::Array4<amrex::Real> const &Sy = face_areas[1]->array(mfi);
+        amrex::Array4<amrex::Real> const &Sz = face_areas[2]->array(mfi);
+
+        // Extract tileboxes for which to loop
+        Box const &trhox = mfi.tilebox(Rhofield[0]->ixType().toIntVect());
+        Box const &trhoy = mfi.tilebox(Rhofield[1]->ixType().toIntVect());
+        Box const &trhoz = mfi.tilebox(Rhofield[2]->ixType().toIntVect());
+
+        amrex::ParallelFor(trhox, trhoy, trhoz,
+
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                if (Sx(i, j, k) <= 0) return;
+
+                Rhox(i, j, k) = (Ey(i, j, k) * ly(i, j, k) - Ey(i, j, k + 1) * ly(i, j, k + 1) +
+                    Ez(i, j + 1, k) * lz(i, j + 1, k) - Ez(i, j, k) * lz(i, j, k)) / Sx(i, j, k);
+
+            },
+
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                if (Sy(i, j, k) <= 0) return;
+
+                Rhoy(i, j, k) = (Ez(i, j, k) * lz(i, j, k) - Ez(i + 1, j, k) * lz(i + 1, j, k) +
+                    Ex(i, j, k + 1) * lx(i, j, k + 1) - Ex(i, j, k) * lx(i, j, k)) / Sy(i, j, k);
+
+            },
+
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                if (Sz(i, j, k) <= 0) return;
+
+                Rhoz(i, j, k) =  (Ex(i, j, k) * lx(i, j, k) - Ex(i, j + 1, k) * lx(i, j + 1, k) +
+                    Ey(i + 1, j, k) * ly(i + 1, j, k) - Ey(i, j, k) * ly(i, j, k)) / Sz(i, j, k);
+
+            }
+        );
+
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            amrex::Gpu::synchronize();
+            wt = amrex::second() - wt;
+            amrex::HostDevice::Atomic::Add( &(*cost)[mfi.index()], wt);
+        }
+    }
+
+}
+
+void FiniteDifferenceSolver::EvolveBCartesianECT (
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Bfield,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& face_areas,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& area_mod,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Rhofield,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Venl,
+    std::array< std::unique_ptr<amrex::iMultiFab>, 3 >& flag_info_cell,
+    std::array< std::unique_ptr<amrex::LayoutData<FaceInfoBox> >, 3 >& borrowing,
+    int lev, amrex::Real const dt ) {
+
+    amrex::LayoutData<amrex::Real> *cost = WarpX::getCosts(lev);
+
+    Venl[0]->setVal(amrex::Real(0));
+    Venl[1]->setVal(amrex::Real(0));
+    Venl[2]->setVal(amrex::Real(0));
```

Comment:
Same comment as elsewhere, this would probably be fine, too (with `_rt` added if there's an implicit conversion warning or something along those lines):
```suggestion
    Venl[0]->setVal(0.);
    Venl[1]->setVal(0.);
    Venl[2]->setVal(0.);
```